### PR TITLE
ci: perform license headers check across entire repo

### PR DESCRIPTION
**Pull request by jgehrcke**
Will also address https://github.com/opstrace/opstrace/issues/73.

todo:
- fix the problems that it finds
- remove the yarn entry points
- document

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -242,6 +242,35 @@ cli-test-s3-latest:
...
+	# show a "no such file or directory" error when the extended glob does not
+	# match a single file in the current directory. That's fine. Note: we start
+	# bash so that we can rely on bash's extended glob behavior. Also note that
+	# `addlicense` does not have well-behaved recursive directory tree walking
+	# behavior (with predictable pattern matching!). That's why we use
+	# find/bash+extglob to control _that_ part. Fail the target when this
+	# technique produces file changes (show the changes, too).
+	find . -not -path '*/.git/*' \
+		-not -path '*/node_modules*' \
+		-not -path '*/kubernetes/src/custom-resources*' \
```
Comment (jgehrcke) [inline Makefile:267]: Note: add exclude patterns here, when new ones come up.

File: Makefile [hunk 2]
```diff
@@ -242,6 +242,35 @@ cli-test-s3-latest:
...
+		-type d -exec bash -c \
+		$$'echo "dir: {}"; shopt -s extglob \n addlicense -c "Opstrace, Inc." -l apache {}/@(*.ts|*.tsx|*.json|*.go|*.css)' \;
```
Comment (jgehrcke) [inline Makefile:269]: Note: add new file extensions here, when they come up.

Comment (jgehrcke) [reply Makefile:269]: Also note: the `{}` is where `find` fills in the path to the current direction (as part of the file system walk).

Comment (jgehrcke) [reply Makefile:269]: Also note: launching a constant number of processes process per directory (bash and addlicense) is probably faster here than launching a constant number of processes per file, assuming that we probably have ~10 times more files than directories.

In CI, this walk passes within ~1 second. On my machine, it's a little slower (5 seconds or so).

Comment (jgehrcke) [reply Makefile:269]: CC @triclambert not sure if you're interested in seeing this ugly piece of code, but here it is :shrug:

File: package.json [hunk 3]
```diff
@@ -96,7 +96,7 @@
       "gofmt -s -w",
       "yarn golint"
     ],
-    "*.{js,css,json,ts}": [
+    "*.{js,css,json,ts,tsx}": [
```
Comment (jgehrcke) [inline package.json:99]: Note: this is just for the commit hook stage. Not quite sure if within a commit this really catches everything. Well. Let's leave this here.

File: Makefile (hunk 2 above)
Comment (triclambert) [reply Makefile:269]: I've been following along.  And steering clear.  😁   

But in all seriousness, thanks for doing this.  It's unfortunate that it's needed, but your solution looks good.

**Comment by jgehrcke** [comment]
main-test-gcp failed as of #62. Merging.

**Pull request merged**